### PR TITLE
feat (docs): add code snippets to 4.0 migration guide

### PR DESCRIPTION
**Pull request by lgrammel**

### CODE REVIEW

File: content/docs/08-troubleshooting/01-migration-guide/01-migration-guide-4-0.mdx [hunk 1]
```diff
@@ -243,18 +473,46 @@ The following telemetry data values have been removed:
...
+```ts filename="AI SDK 3.4"
+import { experimental_createModelRegistry } from 'ai';
+```
+
+```ts filename="AI SDK 4.0"
+import { experimental_createProviderRegistry } from 'ai';
+```
+
+<Note>
+  The model registry is not available any more. Please register providers
```
Comment (nicoalbanese) [inline content/docs/08-troubleshooting/01-migration-guide/01-migration-guide-4-0.mdx:512]:
```suggestion
  The model registry is not available any more. Please [register providers](/docs/reference/ai-sdk-core/provider-registry#setup).
```

File: content/docs/08-troubleshooting/01-migration-guide/01-migration-guide-4-0.mdx [hunk 2]
```diff
@@ -344,14 +719,22 @@ The `simulateReadableStream` function from `ai/test` has been renamed to `chunks
 The AI SDK RSC 3.0 `render` function has been removed.
 Please use the `streamUI` function instead or switch to AI SDK UI.
```
Comment (nicoalbanese) [inline content/docs/08-troubleshooting/01-migration-guide/01-migration-guide-4-0.mdx:720]:
```suggestion
Please use the `streamUI` function instead or [switch to AI SDK UI](/docs/ai-sdk-rsc/migrating-to-ui).
```

**Review by nicoalbanese — APPROVED** [review]

File: content/docs/08-troubleshooting/01-migration-guide/01-migration-guide-4-0.mdx [hunk 3]
```diff
@@ -14,10 +14,18 @@ description: Learn how to upgrade AI SDK 3.4 to 4.0.
 ## Recommended Process
 
 1. [Migrate to AI SDK 3.4](/docs/troubleshooting/migration-guide/migration-guide-3-4).
-2. Resolve all deprecation warnings.
-3. Upgrade to AI SDK 4.0.
-4. _Optional_: Run the [codemod](#codemods) automated migrations.
-5. Follow the breaking changes guide below.
+2. Upgrade to AI SDK 4.0.
```
Comment (nicoalbanese) [inline content/docs/08-troubleshooting/01-migration-guide/01-migration-guide-4-0.mdx:17]: do we want to add in pnpm install ai@latest or 4.0 somwhere?

**Review by nicoalbanese — APPROVED** [review]

File: content/docs/08-troubleshooting/01-migration-guide/01-migration-guide-4-0.mdx (hunk 3 above)
Comment (lgrammel) [reply content/docs/08-troubleshooting/01-migration-guide/01-migration-guide-4-0.mdx:17]: was thinking about how to be do it and opted for the `AI SDK 4.0 package versions` below. we can improve further if you have ideas

**Pull request merged**